cmd/cmd.go

```diff
@@ -382,30 +386,27 @@ func RunGenerate(cmd *cobra.Command, args []string) error {
 
 	// prompts are provided via stdin or args so don't enter interactive mode
 	if len(prompts) > 0 {
-		return generate(cmd, args[0], strings.Join(prompts, " "), wordWrap, format)
+		_, err := generate(cmd, args[0], strings.Join(prompts, " "), nil, wordWrap, format)
+		if err != nil {
+			return err
+		}
+		return nil
 	}
 
 	return generateInteractive(cmd, args[0], wordWrap, format)
 }
 
-type generateContextKey string
-
-func generate(cmd *cobra.Command, model, prompt string, wordWrap bool, format string) error {
+func generate(cmd *cobra.Command, model, prompt string, messages []api.Message, wordWrap bool, format string) (*api.Message, error) {
```

Comment:
On first glance I think we'd want to use a `generateMessage` wrapper instead of stuffing this one, but I'm still going through the PR

Reply:
I moved the generateRequest up a level, makes this a bit cleaner